### ISSUE TITLE
Floodgate Send Floodgate Data Causes Ip Forwarding Error from Bungeecord/Waterfall

### ISSUE THREAD

**Issue by Tydium** [GeyserMC/Geyser#1842]
**Describe the bug**
I setup floodgate on all backend servers to use its api, and the main problem I am having is that It now causes a  Bungeecord IP forward error on joining. Not all the time however. Like 50% of the time it happens so it's a hit or miss if I join my server. I don't know why this happens. All I did was setup floodgate data.

**To Reproduce**

1. Setup Floodgate on all backend servers
2. Set "send floodgate data" to true in bungee configuration
3.Join a server a few times on bedrock and you may or may not get the issue.

**Expected behavior**

I expected the IP Forwarding issue to not be there and just let me join my server (:

**Screenshots / Videos**
Floodgate Video 1: https://twitter.com/tydium/status/1350295734004842500?s=19

Floodgate Video 2: https://twitter.com/tydium/status/1350295919074414592?s=19

**Server Version and Plugins**

Latest Version Waterfall, Geyser Latest Version, Backend Servers run on paper. Server you first join is 1.8.8

**Geyser Dump**

https://dump.geysermc.org/ZSKhl5mZ802ADbXRZuuXCBVsiTkxBDTP

**Minecraft: Bedrock Edition Version**

Latest Versions (I also tried beta versions and this happened) 

**Additional Context**

If you are unable to reproduce the problem please Dm me or ping me on discord (Tydium#4117) I can give you the IP and port to my server and you may test hands on yourself.

In Video 2 it shows what happens in game when connecting to a second backend server once your already in hub. (Please watch that one through the full 8 seconds to see what I fully mean on the issue)

THIS IS NOT A PROBLEM ON JAVA EDITION.

**Comment by Camotoy** [comment]
Please do attach a Geyser dump.

**Comment by Tydium** [comment]
I added a dump to the Dump section of the issue promt.

**Comment by Tydium** [comment]
![Screenshot_20210115-233004_Discord.jpg](https://user-images.githubusercontent.com/67938521/104797108-aaa4df80-5789-11eb-8e24-53762d87b42b.jpg)

This is what happens on the console.

**Comment by Tydium** [comment]
I updated to the latest geyser (I was two commits behind) it still happens.

**Comment by Camotoy** [comment]
Have you tried without any other BungeeCord plugins? Without any other Paper plugins?

**Comment by Tydium** [comment]
Weve been testing each plugin all day. None of them effect it. It still gives me the same error

**Comment by vijaypondini** [comment]
I'm also on a bungeecord network and I face this issue, though I don't need ip forwarding it still does bother me it doesn't work as intended

**Comment by Tydium** [comment]
I have no idea why. I have a feeling this has to do with the key files. I might just reset mine from geyser and floodgate and redo the process to see what happens.

**Comment by Tydium** [comment]
I FIXED IT. I regenerated the encrypted.txt, and both keys. I think the problem mainly was that I forgot to transfer over encrypted.txt. Thanks for the help!

**Comment by Tydium** [comment]
Welp after two days of working perfectly. The issue came back. I have no idea what is causing this at this point. I reset the keys and did everything I know I could do.

**Comment by james090500** [comment]
You said "Setup Floodgate on all backend servers"

You realise floodgate is meant to run on bungee only, and geyser points to bungee only, not any of the servers behind it?

**Comment by Camotoy** [comment]
There is an option for Floodgate to run on backend servers, if it is also on the BungeeCord server. This is generally helpful for API usage.

**Comment by james090500** [comment]
> There is an option for Floodgate to run on backend servers, if it is also on the BungeeCord server. This is generally helpful for API usage.

that's my bad. im out of date

**Comment by Tydium** [comment]
Okay more update on this issue. Its been 20 days since ive mainly had this problem. It now RARELY if not EVER happens. My main notice is that it stopped happening on all my 1.16.5 servers. I have a bungee leading to 1.8.8 and it gets that error once or twice. Another thing that may be related (but is not the cause of the problem) is that I have a plugin that blocks people from joining backend servers. Its called bungeeguard. Basically it does work with Bedrock but like the main issue is showing. It wont accept connections to any 1.8.8 servers because the the "key" from the main server is different when its actually the same. Maybe this could give you insight on what might be happening with the IP forwarding issue. Thought it was something I should share.

In summary I think when a player tries to connect to a 1.8.8 server it will always some type of "Invalid Key" or "invalid id" so it shows the bungee error (and the bungeeguard error) Might have something to do with translation of packets. Im not really a dev so I don't know, but I hope this helps if you are trying to solve what may be happening.

**Comment by KaiKikuchi** [comment]
Hello. I have the same issue for the past 3 months as well. Sometimes while connecting to the server (when I click "Join Server" or when I get moved to another server by the proxy) I get the "If you wish to use IP forwarding, please enable it in your BungeeCord config as well!" message. As it randomly happens, I guess the PaperSpigot server does not receive the player's details from the proxy sometimes. This only happens with Bedrock Edition clients. It has never happened with Java Edition clients.

**Geyser Dump:** https://dump.geysermc.org/27NY5AS49pUQDsyKRbjF8W9XZsUWHycf

I'm using standalone Geyser (latest server-inventory branch, but I had the issue with the master version as well), Waterfall as proxy (online mode, +floodgate), and PaperSpigot 1.8.8 with latest snapshot of ViaVersion and latest version of Floodgate.

**Comment by Tydium** [comment]
Mine seems to be fixed right now. It's extremely rare for me. I would make sure that your key files are copied correctly. And 1.8.8 seems to bruh alot too.

**Comment by KaiKikuchi** [comment]
> I would make sure that your key files are copied correctly.

Yes, I triple checked. Also I'm sure I got a different error and I wasn't able to join at all the first time I didn't copy the key files. I may try resetting them.

I understand it could be 1.8.8 or ViaVersion related, but I can't move away from it. I wish I could just use 1.16.

Thanks for the answer!

**Comment by MrZomka** [comment]
> 
> 
> > I would make sure that your key files are copied correctly.
> 
> Yes, I triple checked. Also I'm sure I got a different error and I wasn't able to join at all the first time I didn't copy the key files. I may try resetting them.
> 
> I understand it could be 1.8.8 or ViaVersion related, but I can't move away from it. I wish I could just use 1.16.
> 
> Thanks for the answer!

Why can't you use 1.16? 🤔

**Comment by KaiKikuchi** [comment]
Resetting the key did not help. The bug still happens, like 20% of times. I'm absolutely sure the keys have been correctly copied around, otherwise I'd get the invalid key or invalid session error. (I tried)

> Why can't you use 1.16? 🤔
The majority of my community is against replacing the current 1.8.8 servers. And overall, 1.8.8 servers perform better.

**Comment by MrZomka** [comment]
> 
> 
> Resetting the key did not help. The bug still happens, like 20% of times. I'm absolutely sure the keys have been correctly copied around, otherwise I'd get the invalid key or invalid session error. (I tried)
> 
> > Why can't you use 1.16? 🤔
> 
> The majority of my community is against replacing the current 1.8.8 servers. And overall, 1.8.8 servers perform better.

Why not 1.12 then? It's as fast as 1.8 in my experience :p

**Comment by Tydium** [comment]
Community only wants 1.8 for pvp. That could be mirrored over to 1.16.

**Comment by Tydium** [comment]
Finally after the longest time. I think floodgate2/geyser2 fixed it. If anything I havent gotten this in months. I'll still let you decide if we should close the issue or not (probably should) My server is now clean of this bug :thumbsup:

**Comment by Konicai** [comment]
Closing as Tydium's issue has been fixed and Floodgate 2.0 has since been released. Join https://discord.gg/GeyserMC if anyone else if having this issue

**Comment by Brasil150** [comment]
This is not fixed. I have the same problem in 1.8 backend.

If I use backend 1.12.2 I don't get the error.

**Comment by Tydium** [comment]
Doesn't Geyser not support certain versions of Java anymore

**Comment by joshwenke** [comment]
We are having this issue intermittently on 1.20.4 and the latest Geyser/Floodate on a proxy setup (Geyser + Floodgate on the proxy, Floodgate on the backend servers)
